source/material_model/visco_plastic.cc

```diff
@@ -394,13 +395,21 @@ namespace aspect
       const ComponentMask volumetric_compositions = get_volumetric_composition_mask();
 
       EquationOfStateOutputs<dim> eos_outputs (this->n_compositional_fields()+1);
+      EquationOfStateOutputs<dim> eos_outputs_all_phases (this->n_compositional_fields()+1+phase_function.n_phase_transitions());
 
       // Loop through all requested points
       for (unsigned int i=0; i < in.temperature.size(); ++i)
         {
-          equation_of_state.evaluate(in, i, eos_outputs);
-
           // First compute the equation of state variables and thermodynamic properties
+          equation_of_state.evaluate(in, i, eos_outputs_all_phases);
+
+          MaterialUtilities::PhaseFunctionInputs<dim> phase_inputs(*this,in,i,eos_outputs_all_phases.densities[0],0);
+
+          compute_equation_of_state_phase_transitions(eos_outputs_all_phases,
```

Comment:
I am not really happy with this name (from the name alone, I find it hard to understand what's happening here). What about something like `phase_average_equation_of_state_outputs`?